api/coverage.py

```diff
@@ -139,7 +139,10 @@ def lookup_and_import_batch(self, batch):
         # id_mapping maps our local identifiers to identifiers the
         # foreign data source will reocgnize.
         id_mapping = self.create_identifier_mapping(batch)
-        foreign_identifiers = id_mapping.keys()
+        if id_mapping:
```

Comment:
I don't have a good way of testing this because it's in a method overridden by MockOPDSImportCoverageProvider. I have tested it in a real situation. This means you don't _have_ to have a mapping; you can return None (which happens by default) and the original identifiers will be used as themselves.